### PR TITLE
Fixes 3465 Add support to platform.sh to our allowed list of staging and development environments

### LINKED ISSUE #3465: Add support to platform.sh to our allowed list of staging and development environments
**Is your feature request related to a problem? Please describe.**
platform.sh is a hosting provider using the following for staging/dev environments: 
**staging-xxx-yyy.xxx-#.platformsh.site
dev-xxx-yyy.xxx-#.platformsh.site**
Using WP Rocket in any of these will count as a license.

**Describe the solution you'd like**
Allow **.platformsh.site** as part of WP Rocket list of staging and development environments.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by jorgemartine00**
Added **.platformsh.site** to $staging list
https://github.com/wp-media/wp-rocket/issues/3465

### CODE REVIEW

**Review by remyperona — APPROVED** [review]

**Comment by remyperona** [comment]
@wp-media/qa ready for QA

**Comment by piotrbak** [comment]
@Tabrisrp @jorgemartine00 This is very misleading:
1. In the title of the `PR` and `Issue` we're mentioning the `.platform.sh`
2. In the `Grooming` section and in the description we have `.platformsh.site`
3. In the code there's `.platform.sh` exclusion
4. I assume that the correct exclusion is the `.platformsh.site`, so we need to correct this.

We need to be strict about that since the change also needs to be done on our website level:
https://github.com/wp-media/wp-rocket/issues/3465#issuecomment-756863123

**Review by piotrbak — CHANGES_REQUESTED** [review]
Please see the above, we need to address the correct exlusion.

File: inc/functions/api.php [hunk 1]
```diff
@@ -129,6 +129,7 @@ function rocket_is_live_site() {
 		'.myftpupload.com',
 		'.dream.press',
 		'.sg-host.com',
+		'.platform.sh',
```
Comment (remyperona) [inline inc/functions/api.php]:
```suggestion
		'.platformsh.site',
```

**Review by remyperona — CHANGES_REQUESTED** [review]

**Comment by remyperona** [comment]
@piotrbak You're right, I reviewed this too fast on Friday. Updated my review with the change needed.

`platform.sh` is the hosting provider, `.platformsh.site` is the domain used.

**Review by remyperona — APPROVED** [review]

**Review by piotrbak — APPROVED** [review]
Worked as expected, adding a comment on the issue that we also need to handle it on the website side.
[testrail-report-115.pdf](https://github.com/wp-media/wp-rocket/files/5881778/testrail-report-115.pdf)

**Pull request merged**